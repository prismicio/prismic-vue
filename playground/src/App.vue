<template>
	<nav>
		<ul>
			<li>
				<router-link to="/">index</router-link>
			</li>
			<li>
				helpers
				<ul>
					<li>
						<router-link to="/helpers/options">options</router-link>
					</li>
					<li>
						<router-link to="/helpers/composition">composition</router-link>
					</li>
				</ul>
			</li>
			<li>
				components
				<ul>
					<li>
						<router-link to="/components/image">image</router-link>
					</li>
					<li>
						<router-link to="/components/embed">embed</router-link>
					</li>
					<li>
						<router-link to="/components/link">link</router-link>
					</li>
					<li>
						<router-link to="/components/text">text</router-link>
					</li>
					<li>
						<router-link to="/components/richtext">richtext</router-link>
					</li>
					<li>
						<router-link to="/components/slicezone">slicezone</router-link>
					</li>
				</ul>
			</li>
		</ul>
	</nav>
	<hr />
	<router-view />
</template>

<script lang="ts">
import { defineComponent } from "vue"

export default defineComponent({
	name: "App",
})
</script>

<style>
body {
	margin: 0;
	padding: 0;
	font-family: sans-serif;
}
</style>
