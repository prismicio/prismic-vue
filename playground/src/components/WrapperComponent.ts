import { defineComponent, h } from "vue"

export const WrapperComponent = defineComponent({
	name: "WrapperComponent",
	setup(_props, { slots }) {
		return () => {
			return h(
				"div",
				{
					class: "wrapperComponent",
				},
				slots.default && slots.default(),
			)
		}
	},
})
