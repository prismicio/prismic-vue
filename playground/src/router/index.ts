import { RouteRecordRaw, createRouter, createWebHistory } from "vue-router";

const routes: Array<RouteRecordRaw> = [
	{
		path: "/",
		name: "Index",
		component: () =>
			import(/* webpackChunkName: "index" */ "../views/index.vue"),
	},

	{
		path: "/helpers/options",
		name: "HelpersOptions",
		component: () =>
			import(
				/* webpackChunkName: "helpers--options" */ "../views/helpers/options.vue"
			),
	},
	{
		path: "/helpers/composition",
		name: "HelpersComposition",
		component: () =>
			import(
				/* webpackChunkName: "helpers--composition" */ "../views/helpers/composition.vue"
			),
	},

	{
		path: "/components/image",
		name: "ComponentsImage",
		component: () =>
			import(
				/* webpackChunkName: "components--image" */ "../views/components/image.vue"
			),
	},
	{
		path: "/components/embed",
		name: "ComponentsEmbed",
		component: () =>
			import(
				/* webpackChunkName: "components--embed" */ "../views/components/embed.vue"
			),
	},
	{
		path: "/components/link",
		name: "ComponentsLink",
		component: () =>
			import(
				/* webpackChunkName: "components--link" */ "../views/components/link.vue"
			),
	},
	{
		path: "/components/text",
		name: "ComponentsText",
		component: () =>
			import(
				/* webpackChunkName: "components--text" */ "../views/components/text.vue"
			),
	},
	{
		path: "/components/richtext",
		name: "ComponentsRichText",
		component: () =>
			import(
				/* webpackChunkName: "components--richtext" */ "../views/components/richtext.vue"
			),
	},
];

const router = createRouter({
	history: createWebHistory(),
	routes,
});

export default router;
