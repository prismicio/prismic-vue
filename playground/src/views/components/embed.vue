<template>
	<div class="componentsEmbed">
		<input v-model="inputText" type="text" />{{ inputText }}
		<prismic-embed :field="null" />
		<prismic-embed :field="empty" />
		<prismic-embed :field="youtube" />
		<prismic-embed :field="twitter" />
		<prismic-embed :field="computed" />
	</div>
</template>

<script lang="ts">
import { defineComponent } from "vue";

import { empty, twitter, youtube } from "../../mocks/embed.json";

export default defineComponent({
	name: "ComponentsEmbed",
	data() {
		return {
			empty,
			youtube,
			twitter,
			inputText: "",
		};
	},
	computed: {
		computed() {
			return {
				...this.youtube,
				html: `${this.youtube.html}<div>${this.inputText}</div>`,
			};
		},
	},
});
</script>
