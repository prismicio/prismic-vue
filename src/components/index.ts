export { PrismicEmbedImpl, PrismicEmbed } from "./PrismicEmbed"
export type { PrismicEmbedProps } from "./PrismicEmbed"

export { usePrismicImage, PrismicImageImpl, PrismicImage } from "./PrismicImage"
export type { UsePrismicImageOptions, PrismicImageProps } from "./PrismicImage"

export { usePrismicLink, PrismicLinkImpl, PrismicLink } from "./PrismicLink"
export type { UsePrismicLinkOptions, PrismicLinkProps } from "./PrismicLink"

export { usePrismicText, PrismicTextImpl, PrismicText } from "./PrismicText"
export type { UsePrismicTextOptions, PrismicTextProps } from "./PrismicText"

export {
	usePrismicRichText,
	PrismicRichTextImpl,
	PrismicRichText,
} from "./PrismicRichText"
export type {
	UsePrismicRichTextOptions,
	PrismicRichTextProps,
} from "./PrismicRichText"

export {
	getSliceComponentProps,
	TODOSliceComponent,
	defineSliceZoneComponents,
	SliceZoneImpl,
	SliceZone,
} from "./SliceZone"
export type {
	DefineComponentSliceComponentProps,
	SliceComponentProps,
	SliceComponentType,
	SliceLikeRestV2,
	SliceLikeGraphQL,
	SliceLike,
	SliceZoneComponents,
	SliceZoneResolverArgs,
	SliceZoneResolver,
	SliceZoneLike,
	SliceZoneProps,
} from "./SliceZone"
