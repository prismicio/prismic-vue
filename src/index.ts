import PrismicRichText from "./PrismicRichText/PrismicRichText.vue"
import PrismicTable from "./PrismicTable/PrismicTable.vue"
import SliceZone from "./SliceZone/SliceZone.vue"

import PrismicEmbed from "./PrismicEmbed.vue"
import PrismicImage from "./PrismicImage.vue"
import PrismicLink from "./PrismicLink.vue"
import PrismicText from "./PrismicText.vue"

export type { SliceZoneProps } from "./SliceZone/SliceZone.vue"
export type { PrismicRichTextProps } from "./PrismicRichText/PrismicRichText.vue"
export type { PrismicTableProps } from "./PrismicTable/PrismicTable.vue"

export type { PrismicEmbedProps } from "./PrismicEmbed.vue"
export type { PrismicImageProps } from "./PrismicImage.vue"
export type { PrismicLinkProps } from "./PrismicLink.vue"
export type { PrismicTextProps } from "./PrismicText.vue"

export {
	SliceZone,
	PrismicEmbed,
	PrismicImage,
	PrismicLink,
	PrismicTable,
	PrismicText,
	PrismicRichText,
}

export { getRichTextComponentProps } from "./PrismicRichText"
export type {
	VueRichTextSerializer,
	RichTextComponentProps,
} from "./PrismicRichText"

export { getTableComponentProps } from "./PrismicTable"

export {
	TODOSliceComponent,
	defineSliceZoneComponents,
	getSliceComponentProps,
} from "./SliceZone"
export type {
	SliceComponentProps,
	SliceComponentType,
	SliceLike,
	SliceLikeGraphQL,
	SliceLikeRestV2,
	SliceZoneComponents,
	SliceZoneLike,
} from "./SliceZone"

export type { PrismicPluginOptions, PrismicPlugin } from "./types"

export { createPrismic } from "./createPrismic"
export { usePrismic, prismicKey } from "./usePrismic"
