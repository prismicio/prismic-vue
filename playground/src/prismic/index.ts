import { createPrismic } from "../../../src"

const prismic = createPrismic({
	endpoint: "200629-sms-hoy",
	linkResolver: (doc) => `/${doc.uid}`,
})

export default prismic
