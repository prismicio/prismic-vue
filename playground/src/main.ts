import { createApp } from "vue";

import App from "./App.vue";
import prismic from "./prismic";
import router from "./router";

createApp(App).use(router).use(prismic).mount("#app");
