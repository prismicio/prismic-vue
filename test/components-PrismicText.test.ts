import { expect, it } from "vitest";

import { RichTextField } from "@prismicio/client";
import { mount } from "@vue/test-utils";
import { markRaw } from "vue";

import { WrapperComponent } from "./__fixtures__/WrapperComponent";
import { richTextFixture } from "./__fixtures__/richText";

import { PrismicTextImpl } from "../src/components";

it("renders rich text field as plain text", () => {
	const wrapper = mount(PrismicTextImpl, {
		props: { field: richTextFixture.en },
	});

	expect(wrapper.html()).toMatchSnapshot();
});

it("renders fallback when rich text is empty", () => {
	const wrapper1 = mount(PrismicTextImpl, {
		props: { field: [] },
	});

	expect(wrapper1.html()).toBe("<div></div>");

	const wrapper2 = mount(PrismicTextImpl, {
		props: { field: [], fallback: "foo" },
	});

	expect(wrapper2.html()).toBe("<div>foo</div>");

	const wrapper3 = mount(PrismicTextImpl, {
		props: { field: null, fallback: "bar" },
	});

	expect(wrapper3.html()).toBe("<div>bar</div>");

	const wrapper4 = mount(PrismicTextImpl, {
		props: { field: undefined, fallback: "baz" },
	});

	expect(wrapper4.html()).toBe("<div>baz</div>");
});

it("uses provided wrapper tag", () => {
	const wrapper = mount(PrismicTextImpl, {
		props: { field: richTextFixture.en, wrapper: "p" },
	});

	expect(wrapper.html()).toMatchSnapshot();
});

it("uses provided wrapper component", () => {
	const wrapper = mount(PrismicTextImpl, {
		props: { field: richTextFixture.en, wrapper: markRaw(WrapperComponent) },
	});

	expect(wrapper.html()).toMatchSnapshot();
});

it("reacts to changes properly", async () => {
	const wrapper = mount(PrismicTextImpl, {
		props: { field: richTextFixture.en },
	});

	const firstRender = wrapper.html();

	await wrapper.setProps({
		field: [{ type: "paragraph", text: "foo", spans: [] }] as RichTextField,
	});

	const secondRender = wrapper.html();

	expect(secondRender).not.toBe(firstRender);
	expect(secondRender).toBe("<div>foo</div>");
});
