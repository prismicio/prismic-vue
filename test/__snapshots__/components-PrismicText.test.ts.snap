// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders rich text field as plain text 1`] = `"<div>Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</div>"`;

exports[`uses provided wrapper component 1`] = `"<div class="wrapperComponent">Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</div>"`;

exports[`uses provided wrapper tag 1`] = `"<p>Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>"`;
